distributedCache/LRUCache.java

```diff
@@ -0,0 +1,48 @@
+package distributedCache;
+
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class LRUCache<K,V> extends Cache<K,V>{
+
+    private final ConcurrentHashMap<K,V> map = new ConcurrentHashMap<>();
+    private final Deque<K> q = new LinkedList<>();
+
+    public LRUCache(int size) {

```

Comment:
nit : is it better to create a map of the size defined when creating this hashmap?